api/v1beta1/machinedeployment_webhook.go

```diff
@@ -214,6 +215,14 @@ func (m *MachineDeployment) validate(old *MachineDeployment) error {
 		)
 	}
 
+	// MachineSet preflight checks that should be skipped could also be set as annotation on the MachineDeployment
+	// since MachineDeployment annotations are synced to the MachineSet.
```

Comment:
Let's add this note to the annotation comment as well, it could be helpful for the users that are not aware of the propagation rules.